realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -172,32 +173,35 @@ private boolean containsDot(String s) {
     }
 
     /**
-     * Test if a field is null. Only works for relationships and RealmLists.
+     * Test if a field is {@code null}. Only works for not required types.
      *
-     * @param fieldName - the field name
-     * @return The query object
-     * @throws java.lang.IllegalArgumentException if field is not a RealmObject or RealmList
+     * @param fieldName - the field name.
+     * @return The query object.
+     * @throws java.lang.IllegalArgumentException if the field is annotated with @{@link io.realm.annotations.Required}
+     * or the field is with the type {@link RealmList}.
      */
     public RealmQuery<E> isNull(String fieldName) {
-        // Currently we only support querying top-level
-        if (containsDot(fieldName)) {
-            throw new IllegalArgumentException("Checking for null in nested objects is not supported.");
-        }
+        long columnIndices[] = getColumnIndices(fieldName, null);
 
         // checking that fieldName has the correct type is done in C++
-        this.query.isNull(columns.get(fieldName));
+        this.query.isNull(columnIndices);
         return this;
     }
 
     /**
-     * Test if a field is not null. Only works for relationships and RealmLists.
+     * Test if a field is not {@code null}. Only works for not required types.
      *
-     * @param fieldName - the field name
-     * @return The query object
-     * @throws java.lang.IllegalArgumentException if field is not a RealmObject or RealmList
+     * @param fieldName - the field name.
+     * @return The query object.
+     * @throws java.lang.IllegalArgumentException if the field is annotated with @{@link io.realm.annotations.Required}
```

Comment:
"If the field is a primitive type, A RealmList or a boxed type wit the @Required annotation.